receiver/apachedruidreceiver/metrics.go

```diff
@@ -0,0 +1,306 @@
+
+			mReceiver.logger.Debug("Starting ServeTLS",
+				zap.String("address", mReceiver.config.Endpoint),
+				zap.String("certfile", mReceiver.config.TLS.CertFile),
+				zap.String("keyfile", mReceiver.config.TLS.KeyFile))
+
+			err := mReceiver.server.ServeTLS(listener, mReceiver.config.TLS.CertFile, mReceiver.config.TLS.KeyFile)
+
+			mReceiver.logger.Debug("Serve TLS done")
+
+			if err != http.ErrServerClosed {
+				mReceiver.logger.Error("ServeTLS failed", zap.Error(err))
+				host.ReportFatalError(err)
+			}
+		}()
+		return nil
+	}
+
+	go func() {
+		defer mReceiver.wg.Done()
+
+		mReceiver.logger.Debug("Starting Serve",
+			zap.String("address", mReceiver.config.Endpoint))
+
+		err = mReceiver.server.Serve(listener)
+
+		mReceiver.logger.Debug("Serve done")
+
+		if err != http.ErrServerClosed {
+			mReceiver.logger.Error("Serve failed", zap.Error(err))
+			host.ReportFatalError(err)
+		}
+	}()
+
+	return nil
+}
+
+func (mReceiver *metricsReceiver) handleRequest(rw http.ResponseWriter, request *http.Request) {
+	if mReceiver.config.BasicAuth != nil {
+		auth := request.Header.Get("Authorization")
+		credentials := fmt.Sprintf("%s:%s", mReceiver.config.BasicAuth.Username, mReceiver.config.BasicAuth.Password)
+		configAuth := fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(credentials)))
+		if auth == "" {
+			rw.WriteHeader(http.StatusUnauthorized)
+			mReceiver.logger.Debug("Got request with no basic auth credentials when they were specified in config, dropping...")
+			return
+		} else if auth != configAuth {
+			rw.WriteHeader(http.StatusUnauthorized)
+			mReceiver.logger.Debug("Got request with incorrect basic auth credentials when they were specified in config, dropping...")
+			return
+		}
+	}
+
+	if request.Method != "POST" {
+		rw.WriteHeader(http.StatusBadRequest)
+		mReceiver.logger.Debug("Receiver server only accepts POST requests", zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	if request.Header.Get("Content-Type") != "application/json" {
+		rw.WriteHeader(http.StatusBadRequest)
+		errMessage := "Content type must be JSON"
+		if _, err := rw.Write([]byte(errMessage)); err != nil {
+			mReceiver.logger.Debug("Error writing response message to Druid", zap.Error(err), zap.String("remote", request.RemoteAddr))
+			return
+		}
+		mReceiver.logger.Debug(errMessage, zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	payload, err := io.ReadAll(request.Body)
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		mReceiver.logger.Debug("Failed to read metrics payload", zap.Error(err), zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	var metrics []datapoint
+	if err = json.Unmarshal(payload, &metrics); err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		mReceiver.logger.Debug("Failed to convert metrics payload from JSON array to golang slice", zap.Error(err), zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	purgedMetrics, service := purgeMetrics(metrics)
```

Comment:
Purge seems like it could mean more than just filtering the received list down to the supported metrics. Maybe a rename to something like "filterMetrics" or "supportedMetrics" or similar.